macros/sql_builders.sql

```diff
@@ -80,79 +72,41 @@
-      {% endfor %}
-      select {{ columns | join (",") }}
-    {% else %}
-      {{ exceptions.raise_compiler_error("Model " ~ node.name ~ " columns must be declared in schema.yml, or it must exist in database") }}
-    {% endif %}
-  {% endif %}
+  {{ dbt_unit_testing.get_columns_sql(node, node.name, node.columns, "Model " ~ node.name ~ " columns must be declared in schema.yml, or it must exist in database") }}
 {% endmacro %}
 
 {% macro fake_source_sql(node) %}
-  {% set source_relation = dbt_utils.get_relations_by_pattern(
-      database=node.database,
-      schema_pattern=node.schema,
-      table_pattern=node.identifier
-  ) %}
-  {% if source_relation | length > 0 %}
-    {%- set source_sql -%}
-      select * from {{ dbt_unit_testing.source_node_to_sql(node) }} where false
-    {%- endset -%}
-    select {{ dbt_unit_testing.quote_and_join_columns(dbt_unit_testing.extract_columns_list(source_sql)) }}
-    from {{ dbt_unit_testing.source_node_to_sql(node) }}
-    where false
-  {% else %}
-    {% if node.columns %}
-      {% set columns = [] %}
-      {% for c in node.columns.values() %}
-        {% do columns.append("cast(null as " ~ (c.data_type if c.data_type is not none else dbt_utils.type_string()) ~ ") as " ~ dbt_unit_testing.quote_identifier(c.name)) %}
-      {% endfor %}
-      select {{ columns | join (",") }}
-    {% else %}
-      {{ exceptions.raise_compiler_error("Source " ~ node.name ~ " columns must be declared in sources.yml, or it must exist in database") }}
-    {% endif %}
-  {% endif %}
+  {{ dbt_unit_testing.get_columns_sql(node, node.identifier, node.columns, "Source " ~ node.name ~ " columns must be declared in sources.yml, or it must exist in database") }}
 {% endmacro %}
 
 {% macro fake_seed_sql(node) %}
-  {% set source_relation = dbt_utils.get_relations_by_pattern(
-      database=node.database,
-      schema_pattern=node.schema,
-      table_pattern=node.name
-  ) %}
-  {% if source_relation | length > 0 %}
-    {%- set source_sql -%}
-      select * from {{ dbt_unit_testing.model_node_to_sql(node) }} where false
-    {%- endset -%}
-    select {{ dbt_unit_testing.quote_and_join_columns(dbt_unit_testing.extract_columns_list(source_sql)) }}
-    from {{ dbt_unit_testing.model_node_to_sql(node) }}
+  {% set columns = {} %}
+  {% if node.config and node.config.column_types %}
+    {% for c in node.config.column_types.keys() %}
+    {% do columns.update({c: {"name" : c, "data_type": node.config.column_types[c]} }) %}
+    {% endfor %}
+  {% endif %}
+  {{ dbt_unit_testing.get_columns_sql(node, node.name, columns, "Seed " ~ node.name ~ " columns must be declared in properties.yml, or it must exist in database") }}
+{% endmacro %}
+
+{% macro get_columns_sql(node, identifier, config_columns, error_message) %}
+  {% set columns = adapter.get_columns_in_relation(api.Relation.create(database=node.database, schema=node.schema, identifier=identifier)) %}
```

Comment:
Here we also need to use the dbt_unit_testing.quote_identifier(identifier).
It will make the snowflake tests pass.

Reply:
I think the problem could be caused by quoting set to False by default for Snowflake.
I just added a commit where I explicitly set quoting to true for this particular source, hope this helps.
Could you run the tests again and see if it works?

Reply:
I now force-pushed a new version of that commit, which ensures to pass the quoting parameters on to Relation.create(), which seems to fix the issue. I have run the snowflake tests using my own snowflake project, and they seem to work now :)

Reply:
It worked as you mentioned. 

Thank you for your patience and effort in contributing to the package.